core/positions/engine.go

```diff
@@ -65,14 +70,54 @@ func New(log *logging.Logger, config Config, marketID string, broker Broker) *En
 	log = log.Named(namedLogger)
 	log.SetLevel(config.Level.Get())
 
-	return &Engine{
-		marketID:     marketID,
-		Config:       config,
-		log:          log,
-		positions:    map[string]*MarketPosition{},
-		positionsCpy: []events.MarketPosition{},
-		broker:       broker,
+	e := &Engine{
+		marketID:         marketID,
+		Config:           config,
+		log:              log,
+		positions:        map[string]*MarketPosition{},
+		positionsCpy:     []events.MarketPosition{},
+		broker:           broker,
+		updatedPositions: map[string]struct{}{},
+	}
+	e.positionUpdated = e.bufferPosition
+	if config.StreamPositionVerbose {
+		e.positionUpdated = e.sendPosition
+	}
+
+	return e
+}
+
+func (e *Engine) FlushPositionEvents(ctx context.Context) {
+	if e.StreamPositionVerbose {
+		return
+	}
+
+	e.sendBufferedPosition(ctx)
+}
+
+func (e *Engine) bufferPosition(ctx context.Context, pos *MarketPosition) {
+	e.updatedPositions[pos.partyID] = struct{}{}
+}
+
+func (e *Engine) sendBufferedPosition(ctx context.Context) {
+	parties := maps.Keys(e.updatedPositions)
+	sort.Strings(parties)
+	evts := make([]events.Event, 0, len(parties))
+
+	for _, v := range parties {
+		// ensure the position exists,
+		// party might have been distressed or something
+		if pos, ok := e.positions[v]; ok {
+			evts = append(evts, events.NewPositionStateEvent(ctx, pos, e.marketID))
+		}
 	}
+
+	e.broker.SendBatch(evts)
+	e.updatedPositions = map[string]struct{}{}
```

Comment:
maybe `e.updatedPositions = make(map[string]struct{}, len(e.updatedPositions))` or something to cut back on some realloc at runtime?

Reply:
addressed